src/base/bittorrent/session.cpp

```diff
@@ -2172,6 +2172,13 @@ bool Session::addTorrent_impl(CreateTorrentParams params, const MagnetUri &magne
     m_addingTorrents.insert(hash, params);
     // Adding torrent to BitTorrent session
     m_nativeSession->async_add_torrent(p);
+    if (params.skipChecking) {
+        lt::torrent_handle handle = m_nativeSession->find_torrent(hash);
+        while (!handle.is_valid()) {
```

Comment:
It's unacceptable solution! What's the point of adding torrents asynchronously if you wait for each to be added?

Reply:
You should apply your logic in the place where added torrents are handled.

Reply:
Afaik qbt doesn't have an option to add multiple torrents in seed mode. Multiple torrents are added async when maybe you load torrents from a watch directory or RSS. But those don't have the skip hash chk option.
Also when adding torrents in seed mode they add quickly. I didn't find any waiting issue. Anyways I'll try the other route.

Reply:
How can you check if a already torrent was added in seed mode? I couldn't find a way.

Reply:
@an0n666 

> You should apply your logic in the place where added torrents are handled.

I think @glassez means you should add your logic here:
`Session::handleAddTorrentAlert`

Reply:
Is it possible to check the torrent add parameter(seed mode) there? If not then there's no point in calling a read piece for any added torrent.

Reply:
Should I just call add_torrent for seed mode and async_add_torrent for all other torrents? Since there's no way to check the flags for async added torrents from the torrent add alert.

Reply:
Looks like just calling read_piece is not enough. On pause/resume or client restart the torrent starts working again without doing a recheck.

Reply:
>Should I just call add_torrent for seed mode and async_add_torrent for all other torrents? 

No, of course!
>Since there's no way to check the flags for async added torrents from the torrent add alert.

How do you think we do this with all the other parameters?
>Looks like just calling read_piece is not enough. On pause/resume or client restart the torrent starts working again without doing a recheck.

Seems you need to check whether some (random) file exists and then read some (random) piece belongs to this file.
If file doesn't exist you can call "force recheck".

Reply:
>Afaik qbt doesn't have an option to add multiple torrents in seed mode.

qBittorrent core has no such limitation.

Reply:
> How do you think we do this with all the other parameters?

The add torrent alert returns the parameters but it doesn't return the hash..how to identify torrent without hash?

Reply:
The seed mode parameter stays in fast resume. So on client restart the torrent will start seeding again. Doesn’t matter if you run recheck or whatnot if file isn’t found. The best solution would be to get rid of this feature.